src/main/java/dan200/computercraft/client/render/MonitorTextureBufferShader.java

```diff
@@ -55,25 +58,29 @@ static void setupUniform( int width, int height, Palette palette, boolean greysc
         MATRIX_BUFFER.rewind();
         OpenGlHelper.glUniformMatrix4( uniformP, false, MATRIX_BUFFER );
 
-        OpenGlHelper.glUniform1i( uniformWidth, width );
-        OpenGlHelper.glUniform1i( uniformHeight, height );
+        GL15.glBindBuffer( GL31.GL_UNIFORM_BUFFER, uniformBuffer );
+        uboBuffer = GL15.glMapBuffer( GL31.GL_UNIFORM_BUFFER, GL15.GL_WRITE_ONLY, uboBuffer );
+        uboBuffer.putInt( UNIFORM_OFFSETS.get( 0 ), width );
```

Comment:
The whole `.get(_)` everywhere is _kinda_ gross. Ughr, I'd prefer if we had individual offset/stride fields, but then I realise that makes querying them more ugly.

Reply:
> Ughr, I'd prefer if we had individual offset/stride fields, but then I realise that makes querying them more ugly.

You'd effectively just be moving all the `.get(_)`s somewhere else. LWJGL without a wrapper just isn't pretty, I guess we have to live with that.

Reply:
Oh I know. But the ugliness would be in one place, rather than two.

Reply:
Fair enough, done.